### DISCUSSION TITLE
Annoying flash of completion dots

### DISCUSSION THREAD

**Discussion by staab** [ohmyzsh/ohmyzsh#9922] [Help] [answered]
I'm experiencing a brief flash of completion dots in zsh having installed heroku's completions. I imagine it's because the completion tool takes greater than some tiny threshhold (100ms?) to finish. However, I have `COMPLETION_WAITING_DOTS="false"` in my .zshrc, so I imagine it shouldn't be showing dots at all. Anyway, you can see a short video of what I'm seeing [here](https://user-images.githubusercontent.com/2494842/119208709-42784a00-ba58-11eb-93d5-fef2256d7b8d.mov), I'd welcome any ideas for how to fix it.

**Comment by mcornella** [comment, marked as answer]
Can you post your zshrc file? You shouldn't see that if `COMPLETION_WAITING_DOTS` is not set to true.

**Reply by staab** [reply]
```
export ZSH="/Users/jstaab/.oh-my-zsh"

ZSH_THEME="robbyrussell"
DISABLE_AUTO_TITLE="true"
COMPLETION_WAITING_DOTS="false"
HISTSIZE=50000
SAVEHIST=10000

plugins=(git python pip osx)

source $ZSH/oh-my-zsh.sh

# User configuration

export EDITOR='kak'

# https://unix.stackexchange.com/a/371453/85459
setopt GLOB_DOTS

# https://github.com/robbyrussell/oh-my-zsh/issues/2537
unsetopt share_history

# Path

PATH="$PATH:/Users/jstaab/Documents/cateract"
PATH="$PATH:/Users/jstaab/Documents/psql-tmux/bin"

# Nvm

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

# Compiler flags and package config

export LDFLAGS="$LDFLAGS -L/opt/homebrew/opt/openssl@1.1/lib"
export CPPFLAGS="$CPPFLAGS -I/opt/homebrew/opt/openssl@1.1/include"
export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/opt/homebrew/opt/openssl@1.1/lib/pkgconfig"

export LDFLAGS="$LDFLAGS -L/opt/homebrew/opt/zlib/lib"
export CPPFLAGS="$CPPFLAGS -I/opt/homebrew/opt/zlib/include"
export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/opt/homebrew/opt/zlib/lib/pkgconfig"

# heroku autocomplete setup
HEROKU_AC_ZSH_SETUP_PATH=/Users/jstaab/Library/Caches/heroku/autocomplete/zsh_setup && test -f $HEROKU_AC_ZSH_SETUP_PATH && source $HEROKU_AC_ZSH_SETUP_PATH;
```

If I remove the very last line the problem disappears.

**Reply by mcornella** [reply]
That's a heroku problem then. You don't need `COMPLETION_WAITING_DOTS=false`. See this issue https://github.com/heroku/cli/issues/957#issuecomment-409328668.

**Reply by staab** [reply]
Ah, thank you, I failed to find that the first time around.
